File: crossecore-java/src/java_/ASTNodeBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class ASTNodeBase 
extends org.eclipse.emf.ecore.impl.MinimalEObjectImpl.Container implements ASTNode
{
	private com.crossecore.ocl.OrderedSet<Comment> _comments;
	
	public com.crossecore.ocl.OrderedSet<Comment> getComments()
	{
		if(_comments==null){
			_comments = new com.crossecore.ocl.OrderedSet<Comment>(Comment.class, this, Java_PackageImpl.ASTNODE__COMMENTS, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.ASTNODE__COMMENTS);
		}
		return _comments;
	
	}
	private CompilationUnit _originalCompilationUnit;
	public CompilationUnit getOriginalCompilationUnit()
	{
		
			return _originalCompilationUnit;
		
		
	}
	public void setOriginalCompilationUnit(CompilationUnit value){
		CompilationUnit oldvalue = _originalCompilationUnit;
		_originalCompilationUnit = value;
		if (eNotificationRequired()){
			eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.ASTNODE__ORIGINAL_COMPILATION_UNIT , oldvalue, value));
		}
	}
	private ClassFile _originalClassFile;
	public ClassFile getOriginalClassFile()
	{
		
			return _originalClassFile;
		
		
	}
	public void setOriginalClassFile(ClassFile value){
		ClassFile oldvalue = _originalClassFile;
		_originalClassFile = value;
		if (eNotificationRequired()){
			eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.ASTNODE__ORIGINAL_CLASS_FILE , oldvalue, value));
		}
	}

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.ASTNODE;
	}
	
	
	public NotificationChain basicSetOriginalCompilationUnit(CompilationUnit newobj, NotificationChain msgs) {
		CompilationUnit oldobj = _originalCompilationUnit;
		_originalCompilationUnit = newobj;
		if (eNotificationRequired()) {
			ENotificationImpl notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.ASTNODE__ORIGINAL_COMPILATION_UNIT, oldobj, newobj);
			if (msgs == null){
				msgs = notification;
			}
			else{
				msgs.add(notification);
			}
		}
		return msgs;
	}
	public NotificationChain basicSetOriginalClassFile(ClassFile newobj, NotificationChain msgs) {
		ClassFile oldobj = _originalClassFile;
		_originalClassFile = newobj;
		if (eNotificationRequired()) {
			ENotificationImpl notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.ASTNODE__ORIGINAL_CLASS_FILE, oldobj, newobj);
			if (msgs == null){
				msgs = notification;
			}
			else{
				msgs.add(notification);
			}
		}
		return msgs;
	}
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.ASTNODE__COMMENTS:
				return getComments();
			case Java_PackageImpl.ASTNODE__ORIGINAL_COMPILATION_UNIT:
				return getOriginalCompilationUnit();
			case Java_PackageImpl.ASTNODE__ORIGINAL_CLASS_FILE:
				return getOriginalClassFile();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.ASTNODE__COMMENTS:
				getComments().clear();
				getComments().addAll((java.util.Collection<? extends Comment>) newValue);
				return;
			case Java_PackageImpl.ASTNODE__ORIGINAL_COMPILATION_UNIT:
				setOriginalCompilationUnit((CompilationUnit) newValue);
				return;
			case Java_PackageImpl.ASTNODE__ORIGINAL_CLASS_FILE:
				setOriginalClassFile((ClassFile) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.ASTNODE__COMMENTS:
				return _comments != null && !_comments.isEmpty();
			case Java_PackageImpl.ASTNODE__ORIGINAL_COMPILATION_UNIT:
				return getOriginalCompilationUnit() != null; //single, volatile
			case Java_PackageImpl.ASTNODE__ORIGINAL_CLASS_FILE:
				return getOriginalClassFile() != null; //single, volatile
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-typescript/java_/ASTNodeBase.ts
import {OrderedSet} from "ecore/OrderedSet";
import {EClass} from "ecore/EClass";
import {Set} from "ecore/Set";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {AbstractCollection} from "ecore/AbstractCollection";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {CompilationUnit} from "./CompilationUnit";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {Comment} from "./Comment";
import {ClassFile} from "./ClassFile";
import {ASTNode} from "./ASTNode";
		
			export class ASTNodeBase
			extends BasicEObjectImpl
			implements ASTNode
			{
				private _comments:OrderedSet<Comment> = null;
				
				get comments():OrderedSet<Comment>{
					if(this._comments===null){
						this._comments = new OrderedSet<Comment>(this, Java_PackageLiterals.ASTNODE_COMMENTS, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.ASTNODE_COMMENTS);
							
					}
					return this._comments;
					
				}
				
				
				private _originalCompilationUnit:CompilationUnit = null;
				get originalCompilationUnit():CompilationUnit{
				
					return this._originalCompilationUnit;
				}
				set originalCompilationUnit(value:CompilationUnit) {
					let oldvalue = this._originalCompilationUnit;
					this._originalCompilationUnit = value;
					if (this.eNotificationRequired()){
						this.eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageLiterals.ASTNODE_ORIGINALCOMPILATIONUNIT , oldvalue, value));
					}
				}
				private _originalClassFile:ClassFile = null;
				get originalClassFile():ClassFile{
				
					return this._originalClassFile;
				}
				set originalClassFile(value:ClassFile) {
					let oldvalue = this._originalClassFile;
					this._originalClassFile = value;
					if (this.eNotificationRequired()){
						this.eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageLiterals.ASTNODE_ORIGINALCLASSFILE , oldvalue, value));
					}
				}

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return ASTNodeBase.eStaticClass;
				}
			
			
				public basicSetOriginalCompilationUnit(newobj:CompilationUnit, msgs:NotificationChain):NotificationChain {
					let oldobj = this._originalCompilationUnit;
					this._originalCompilationUnit = newobj;
					if (this.eNotificationRequired()) {
						let notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageLiterals.ASTNODE_ORIGINALCOMPILATIONUNIT, oldobj, newobj);
						if (msgs == null){
							msgs = notification;
						}
						else{
							msgs.add(notification);
						}
					}
					return msgs;
				}
				public basicSetOriginalClassFile(newobj:ClassFile, msgs:NotificationChain):NotificationChain {
					let oldobj = this._originalClassFile;
					this._originalClassFile = newobj;
					if (this.eNotificationRequired()) {
						let notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageLiterals.ASTNODE_ORIGINALCLASSFILE, oldobj, newobj);
						if (msgs == null){
							msgs = notification;
						}
						else{
							msgs.add(notification);
						}
					}
					return msgs;
				}
				
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.ASTNODE_COMMENTS:
							return this.comments;
						case Java_PackageLiterals.ASTNODE_ORIGINALCOMPILATIONUNIT:
							return this.originalCompilationUnit;
						case Java_PackageLiterals.ASTNODE_ORIGINALCLASSFILE:
							return this.originalClassFile;
					}
					//return this.eGetFromBasicEObjectImpl(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
						case Java_PackageLiterals.ASTNODE_COMMENTS:
							this.comments.clear();
							this.comments.addAll(newValue);
							return;
						case Java_PackageLiterals.ASTNODE_ORIGINALCOMPILATIONUNIT:
							this.originalCompilationUnit = <CompilationUnit> newValue;
							return;
						case Java_PackageLiterals.ASTNODE_ORIGINALCLASSFILE:
							this.originalClassFile = <ClassFile> newValue;
							return;
					}
					super.eSet_number_any(featureID, newValue);
				}

				
			}
			
